src/freenas/usr/local/bin/snmp-agent.py

```diff
@@ -413,9 +414,9 @@ def get_list_of_zvols():
     zvols = set()
     root_dir = '/dev/zvol/'
     with contextlib.suppress(FileNotFoundError):  # no zvols
-        for zpool in pathlib.Path(root_dir).iterdir():
-            for zvol in filter(lambda x: '@' not in x.name, zpool.iterdir()):
-                zvol_normalized = zvol.as_posix().removeprefix(root_dir)
+        for zvol in iglob(root_dir + '**', recursive=True):
```

Comment:
iglob is significantly slower than doing `os.scandir`, for example. It's why I used `Path.iterdir()` here. In python3.12, Path.iterdir() will begin to use os.scandir underneath the hood.

Reply:
Could even simplify to using `os.walk` since it uses scandir as well. (NOTE: I haven't tested this :sweat_smile: )
```
for r, d, f in os.walk(root_dir):
    for _f in filter(lambda x: '@' not in x, f):
        zvols.add(os.path.join(r, _f).removeprefix(root_dir).replace('+', ' '))